airbyte-integrations/connectors/source-mysql-v2/src/test/kotlin/io/airbyte/integrations/source/mysql/MysqlCdcIntegrationTest.kt

```diff
@@ -26,13 +28,57 @@ import io.airbyte.protocol.models.v0.SyncMode
 import io.github.oshai.kotlinlogging.KotlinLogging
 import java.sql.Connection
 import java.sql.Statement
+import org.junit.jupiter.api.Assertions.assertEquals
 import org.junit.jupiter.api.BeforeAll
 import org.junit.jupiter.api.Test
 import org.junit.jupiter.api.Timeout
 import org.testcontainers.containers.MySQLContainer
 
 class MysqlCdcIntegrationTest {
 
+    @Test
+    fun testCheck() {
+        val run1: BufferingOutputConsumer = CliRunner.source("check", config(), null).run()
+
+        assertEquals(run1.messages().size, 1)
+        assertEquals(
+            run1.messages().first().connectionStatus.status,
+            AirbyteConnectionStatus.Status.SUCCEEDED
+        )
+
+        MysqlContainerFactory.exclusive(
+                imageName = "mysql:8.0",
+                MysqlContainerFactory.WithCdcOff,
+            )
+            .use { nonCdcDbContainer ->
+                {
+                    val invalidConfig: MysqlSourceConfigurationJsonObject =
+                        MysqlContainerFactory.config(nonCdcDbContainer).apply {
+                            setCursorMethodValue(CdcCursor())
+                        }
+
+                    val nonCdcConnectionFactory =
+                        JdbcConnectionFactory(MysqlSourceConfigurationFactory().make(invalidConfig))
+
+                    provisionTestContainer(nonCdcDbContainer, nonCdcConnectionFactory)
+
+                    val run2: BufferingOutputConsumer =
+                        CliRunner.source("check", invalidConfig, null).run()
+
+                    val messageInRun2 =
+                        run2
+                            .messages()
+                            .filter { it.type == AirbyteMessage.Type.CONNECTION_STATUS }
+                            .first()
+
+                    assertEquals(
+                        AirbyteConnectionStatus.Status.FAILED,
+                        messageInRun2.connectionStatus.status
+                    )
+                }
+            }
+    }
```

Comment:
This is out of scope of this PR but I'm thinking that this should be a unit test instead, or at least also. This will probably require extending the `SelectQuerier` interface to perform `Statement::execute` calls, which will be mocked in the unit test. 

I'm going to think about this some more and this doesn't block this PR.

Reply:
I'd imagine if we had a unit test for this, it would be, mock sql statements return true or false and assert check successful or not. And we'd verify the sql statements are expected sql statement (in this case since it's all static there isn't much to verify here)

The actual content of the sql statements would have to be verified via integration test via a real test container - but that can also be done by manual one time verification on local db since it's not going to be changed. Is that the reason to get rid of the integration test?